lib/statsd.js

```diff
@@ -107,6 +107,34 @@ Client.prototype.timing = function (stat, time, sampleRate, tags, callback) {
   this.sendAll(stat, time, 'ms', sampleRate, tags, callback);
 };
 
+/**
+ * Function that records the time to run a given function
+ * and sends the recorded time (in milliseconds)
+ * @param func {Function} The function to run
+ * @param metricsArgs {Object} The arguments to be used by Client.prototype.timing
```

Comment:
Can you switch this to use the flat list of parameters so this matches other function interfaces. Where we don't need to provide `time {Number}`. I do wonder if `func` should be the first or second parameter since all the other functions have the `stat` name as the first one I think.

```javascript
  * @param stat {String|Array} The stat(s) to send
  * @param sampleRate {Number=} The Number of times to sample (0 to 1). Optional.
  * @param tags {Array=} The Array of tags to add to metrics. Optional.
  * @param callback {Function=} Callback when message is done being delivered. Optional.
```

Reply:
Sure thing 👍  

However, if this uses flat list of parameters rather than an object, then it's probably best to have the `func` as the first parameter so that the metrics args are at least grouped together in the order that they usually appear in other functions, especially since `func` is a non-optional argument here (whereas some of the metrics args are)